### PR TITLE
[SPARK-41433][CONNECT] Make Max Arrow BatchSize configurable

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Make Max Arrow BatchSize configurable

### Why are the changes needed?
make batchsize configurable

### Does this PR introduce _any_ user-facing change?
yes, one new configration

### How was this patch tested?
existing tests

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

**Comment by zhengruifeng** [comment]
cc @grundprinzip @hvanhovell

**Review by grundprinzip — APPROVED** [review]

**Review by dongjoon-hyun — COMMENTED** [review]
+1, late LGTM.

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala [hunk 1]
```diff
@@ -34,4 +35,12 @@ private[spark] object Connect {
       .version("3.4.0")
       .stringConf
       .createOptional
+
+  val CONNECT_GRPC_ARROW_MAX_BATCH_SIZE =
+    ConfigBuilder("spark.connect.grpc.arrow.maxBatchSize")
+      .doc("When using Apache Arrow, limit the maximum size of one arrow batch that " +
+        "can be sent from server side to client side.")
```
[the remark this answers is not in the document]
Comment (zhengruifeng) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala:42]: got it, will update this doc. Thanks!

**Comment by amaliujia** [comment]
+1 late LGTM

**Pull request closed without merging**